src/OpenTelemetry.Sampler.AWS/Clock.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="Clock.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Sampler.AWS;
+
+// A time keeper for the purpose of this sampler.
+internal sealed class Clock
```

Comment:
What is a benefit for using this additional abstraction layer, instread of just using directly DateTime.UtcNow?

Reply:
The sampler does quite a few things based on time, like the rules cache freshness and rate limited sampling (which would come in the next PR. The DateTime APIs can be used for such logic, but I think for testing the functionality, I need a mocked clock which can be passed in as a `Clock` type.